SimPEG/survey.py

```diff
@@ -261,17 +267,26 @@ def _source_list_validator(self, change):
         value = change["value"]
         if len(set(value)) != len(value):
             raise Exception("The source_list must be unique")
-        self._sourceOrder = dict()
+
         [self._sourceOrder.setdefault(src._uid, ii) for ii, src in enumerate(value)]
 
     # TODO: this should be private
     def getSourceIndex(self, sources):
-        if type(sources) is not list:
+        if not isinstance(sources, list):
             sources = [sources]
 
-        for src in sources:
+        for ii, src in enumerate(sources):
             if getattr(src, "_uid", None) is None:
                 raise KeyError("Source does not have a _uid: {0!s}".format(str(src)))
+
+            # Set default value for _sourceOrder again here, even though it's already
+            # done in _source_list_validator. This is because on dask yarn cluster,
+            # _source_list_validator does not always run on all workers. If this line
+            # is not included, the following exception is often thrown when running
+            # inversion on a dask cluster:
+            # KeyError('Some of the sources specified are not in this survey. [None]')
+            self._sourceOrder.setdefault(src._uid, ii)
```

Comment:
honestly, I would lean towards just getting rid of the `_sourceOrder` property all together-- it doesn't seem to be used anywhere, it's "private" so it seems like it's always going to be the default here